### PR TITLE
[doc] Update open-source docs to make getting started easier

### PR DESCRIPTION
**Pull request by kvargha**
## Summary

I updated the Venice Workspace Setup page on the open-source docs with information that I wish I knew when I first got started. This should make it easier for external contributors to get setup.

## How was this PR tested?

N/A

## Does this PR introduce any user-facing changes?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: docs/dev_guide/how_to/workspace_setup.md [hunk 1]
```diff
@@ -7,16 +7,31 @@ permalink: /docs/dev_guide/how_to/workspace_setup
 ---
 
 # Venice Workspace Setup
+We recommend using a Unix-based environment for development, such as Linux or macOS.
+If you're on Windows, we recommend using [WSL](https://learn.microsoft.com/en-us/windows/wsl/install).
```
Comment (adamxchen) [inline docs/dev_guide/how_to/workspace_setup.md:11]: wsl or wsl2?

Comment (kvargha) [reply docs/dev_guide/how_to/workspace_setup.md:11]: I can update it to wsl2. I've heard the experience is much better.

File: docs/dev_guide/how_to/workspace_setup.md [hunk 2]
```diff
@@ -7,16 +7,31 @@ permalink: /docs/dev_guide/how_to/workspace_setup
...
 ```shell
-git clone https://github.com/linkedin/venice.git
+git clone https://github.com/${githubUsername}/venice.git
+cd venice
+git remote add upstream https://github.com/linkedin/venice.git
+git fetch upstream
 ```
 
+## Setting up Java
+We use Java 17 for development. You can download it [here](https://www.oracle.com/java/technologies/javase/jdk17-archive-downloads.html).
```
Comment (adamxchen) [inline docs/dev_guide/how_to/workspace_setup.md:25]: nit: I personally still use 11 and we have been using 17 in production but i don't know if that's every component of venice or just venice-server.

**Review by adamxchen — APPROVED** [review]
Thanks for updating the doc!

File: docs/dev_guide/how_to/workspace_setup.md (hunk 2 above)
Comment (adamxchen) [reply docs/dev_guide/how_to/workspace_setup.md:25]: Feel free to drop this comment

**Pull request merged**